kartothek/serialization/_generic.py

```diff
@@ -268,23 +268,26 @@ def _ensure_type_stability(array_like, value, strict_date_types, require_ordered
     array_like, array_value_type = _handle_categorical_data(array_like, require_ordered)
     array_like, array_value_type = _handle_null_arrays(array_like, value_dtype)
 
-    type_comp = {value_dtype.kind, array_value_type.kind}
+    type_comp = (value_dtype.kind, array_value_type.kind)
 
     compatible_types = [
         # UINT and INT
-        {"u", "i"},
+        ("u", "i"),
+        ("i", "u"),
         # various string kinds
-        {"S", "O"},
-        {"U", "O"},
+        ("S", "O"),
+        ("U", "O"),
+        # bool w/ Nones
+        ("b", "O"),
```

Comment:
I suppose you are using tuples here instead of sets because the order matters ? i.e. `bool, obj` is valid but `obj, bool` isn't?
Is this also the case for `S, O` and `U, O` ?

Reply:
It sadly does. For example, if you would use a tuple, you could query a string (object) column with a bool. So you want to only be fuzzy in one direction.

Reply:
> Is this also the case for S, O and U, O ?
I don't think so. @fjetter ?

Reply:
Ah, the tests found some. I'll have a look.

Reply:
Frankly, I don't care much about `S` types since they are already deprecated by NumPy. They are, however, tested by hypothesis. Whatever makes the tests green :)

I was wondering about the order as well. I was hoping it was agnostic to the order but the case of bool/none alone make a case for a relevant ordering, doesn't it?

valid (O, b)
```
array: [True, False, None] 
value: False
```

invalid (b, O)
```
[True, True, False] 
"ARandomString"
```

Reply:
```

Reply:
> I was wondering about the order as well. I was hoping it was agnostic to the order but the case of bool/none alone make a case for a relevant ordering, doesn't it?

Correct, that's exactly the reason for making this part ordered.

Reply:
How about these cases?

O, b
```
array: ["ARandomString"]
value: True
```

b, O
```
array: [True, False]
value: None
```

Reply:
> O, b

covered by the tests and forbidden (see `test_raise_on_type` with `(True, "True")`).

Reply:
I'll fix the other one.